Analysis/src/SignalObjectSelector.cpp

```diff
@@ -38,10 +38,20 @@ SignalObjectSelector::SignalObjectSelector(SignalMode _mode) : mode(_mode)
         throw analysis::exception("Signal Mode for SignalObjectSelector constructor not supported");
 }
 
-TauIdDiscriminator SignalObjectSelector::GetTauVSjetDiscriminator() const
+std::pair<TauIdDiscriminator, DiscriminatorWP> SignalObjectSelector::GetTauVSjetDiscriminator() const
 {
-    return discriminator;
+    static const std::pair<TauIdDiscriminator, DiscriminatorWP> tauVSjetDiscriminator =
```

Comment:
TauID discriminator depends on the SignalMode => should be a map: mode -> (discr, wp)
Private variable `discriminator` should no longer exist => remove it to avoid confusions in the future.